### ISSUE TITLE
Matching anything beneath `apoc` calls triggers a Neo.DatabaseError.Statement.ExecutionFailed

### ISSUE THREAD

**Issue by philastrophist** [neo4j/neo4j#12589]
- Neo4j version: Desktop1.3.4, driver4.1.0 Enterprise
- Operating system: Windows 10 19041
- API/Driver: Cypher in Neo4jDesktop 1.3.4

## Expected behavior
Execution of query will produce either "no records" or a table of records.
### This query executes fine:
```
MATCH (l1single0:L1Single {fname: "data/single_1002793.fit"})
WITH DISTINCT l1single0 as seed
// match something
MATCH (:Target)-[*]->(seed)
// Get whole tree
OPTIONAL MATCH p1=(seed)<-[*]-(n1)
OPTIONAL MATCH p2=(seed)-[*]->(n2)
WITH collect(p2) as p2s, collect(p1) as p1s, seed
CALL apoc.convert.toTree(p1s+p2s) yield value
RETURN seed, value
ORDER BY seed.id
```
### Moving *any* MATCH under the `apoc` call will result in an ExecutionFailure
```
MATCH (l1single0:L1Single {fname: "data/single_1002793.fit"})
WITH DISTINCT l1single0 as seed
// Get whole tree
OPTIONAL MATCH p1=(seed)<-[*]-(n1)
OPTIONAL MATCH p2=(seed)-[*]->(n2)
WITH collect(p2) as p2s, collect(p1) as p1s, seed
CALL apoc.convert.toTree(p1s+p2s) yield value
// match something
MATCH (:Target)-[*]->(seed)
RETURN seed, value
ORDER BY seed.id
```

## Actual behavior
Exception is raised: Neo.DatabaseError.Statement.ExecutionFailed
In the neo4j.log there is a null reference apparently:
```
2020-09-24 10:49:34.279+0000 ERROR [o.n.b.r.s.i.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.Statement.ExecutionFailed]: null, reference 44079ff6-9aeb-45f2-be50-b3d05971769e.
org.neo4j.fabric.executor.FabricException
	at org.neo4j.fabric.executor.Exceptions.transform(Exceptions.java:51)
	at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:69)
	at org.neo4j.fabric.bolt.BoltQueryExecutionImpl$QueryExecutionImpl.request(BoltQueryExecutionImpl.java:179)
	at org.neo4j.bolt.runtime.AbstractCypherAdapterStream.handleRecords(AbstractCypherAdapterStream.java:104)
	at org.neo4j.bolt.v3.messaging.ResultHandler.onPullRecords(ResultHandler.java:41)
	at org.neo4j.bolt.v4.messaging.PullResultConsumer.consume(PullResultConsumer.java:42)
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine$State.consumeResult(TransactionStateMachine.java:511)
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine$State$2.streamResult(TransactionStateMachine.java:355)
	at org.neo4j.bolt.runtime.statemachine.impl.TransactionStateMachine.streamResult(TransactionStateMachine.java:94)
	at org.neo4j.bolt.v4.runtime.InTransactionState.processStreamResultMessage(InTransactionState.java:73)
	at org.neo4j.bolt.v4.runtime.AbstractStreamingState.processUnsafe(AbstractStreamingState.java:49)
	at org.neo4j.bolt.v4.runtime.InTransactionState.processUnsafe(InTransactionState.java:60)
	at org.neo4j.bolt.v3.runtime.FailSafeBoltStateMachineState.process(FailSafeBoltStateMachineState.java:48)
	at org.neo4j.bolt.runtime.statemachine.impl.AbstractBoltStateMachine.nextState(AbstractBoltStateMachine.java:143)
	at org.neo4j.bolt.runtime.statemachine.impl.AbstractBoltStateMachine.process(AbstractBoltStateMachine.java:91)
	at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:90)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.lambda$enqueue$0(DefaultBoltConnection.java:148)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatchInternal(DefaultBoltConnection.java:237)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:172)
	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:162)
	at org.neo4j.bolt.runtime.scheduling.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:246)
	at org.neo4j.bolt.runtime.scheduling.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$3(ExecutorBoltScheduler.java:229)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.ImmutableCollections$List12.<init>(ImmutableCollections.java:377)
	at java.base/java.util.List.of(List.java:1038)
	at org.neo4j.fabric.stream.Records.of(Records.java:49)
	at org.neo4j.fabric.stream.QuerySubject$RecordQuerySubscriber.onRecordCompleted(QuerySubject.java:270)
	at org.neo4j.cypher.internal.runtime.pipelined.operators.ProduceResultOperator.produceOutput(ProduceResultOperator.scala:185)
	at org.neo4j.cypher.internal.runtime.pipelined.operators.ProduceResultOperator.produceOutputWithProfile(ProduceResultOperator.scala:161)
	at org.neo4j.cypher.internal.runtime.pipelined.operators.ProduceResultOperator$OutputOOperatorState.prepareOutput(ProduceResultOperator.scala:143)
	at org.neo4j.cypher.internal.runtime.pipelined.operators.OutputOperatorState.prepareOutputWithProfile(OutputOperator.scala:56)
	at org.neo4j.cypher.internal.runtime.pipelined.operators.OutputOperatorState.prepareOutputWithProfile$(OutputOperator.scala:45)
	at org.neo4j.cypher.internal.runtime.pipelined.operators.ProduceResultOperator$OutputOOperatorState.prepareOutputWithProfile(ProduceResultOperator.scala:123)
	at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeOutputOperator(PipelineTask.scala:76)
	at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeWorkUnit(PipelineTask.scala:54)
	at org.neo4j.cypher.internal.runtime.pipelined.Worker.executeTask(Worker.scala:135)
	at org.neo4j.cypher.internal.runtime.pipelined.Worker.workOnQuery(Worker.scala:91)
	at org.neo4j.cypher.internal.runtime.pipelined.execution.CallingThreadExecutingQuery.request(CallingThreadExecutingQuery.scala:41)
	at org.neo4j.cypher.internal.PipelinedRuntime$PipelinedRuntimeResult.request(PipelinedRuntime.scala:426)
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:99)
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.request(ClosingExecutionResult.scala:145)
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.doRequest(QuerySubject.java:184)
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.request(QuerySubject.java:167)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.request(Operators.java:2089)
	at reactor.core.publisher.StrictSubscriber.request(StrictSubscriber.java:138)
	at org.neo4j.fabric.stream.Rx2SyncStream$RecordSubscriber.request(Rx2SyncStream.java:129)
	at org.neo4j.fabric.stream.Rx2SyncStream.maybeRequest(Rx2SyncStream.java:91)
	at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:50)
	... 25 more

```

**Comment by mnd999** [comment]
Thanks for the report. Please try prefixing your query with `CYPHER RUNTIME=slotted` and see if you get a better result.

**Comment by pontusmelke** [comment]
@philastrophist thanks for reporting this. This issue should have been fixed, could you please try with 4.1.3?

Regards,
Pontus

**Comment by mnd999** [comment]
We haven't heard anything back on this issue for a while and we believe it to be fixed, so will close for now. Please feel free to reopen if it's still an issue.
